From 6649359080b193204f62933b2f2883dd0f53c5d6 Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@redhat.com>
Date: Fri, 12 Sep 2014 01:56:58 -0400
Subject: [drm] dp: Staticize a couple of DP utility functions

Message-id: <1410487019-24661-21-git-send-email-airlied@redhat.com>
Patchwork-id: 93932
O-Subject: [RHEL7.1 BZ1140440 PATCH 20/21] drm/dp: Staticize a couple of DP utility functions
Bugzilla: 1140440
RH-Acked-by: Jerome Glisse <jglisse@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Rob Clark <rclark@redhat.com>

From: Thierry Reding <treding@nvidia.com>

bugzilla: 1140440
upstream: 8fa6a4255e80537774126440a056c8e4fa7e30a0

sparse complains about these functions missing a prototype, but looking
closer they aren't (and likely not supposed to be) used outside of this
source file, so they can be made static.

Signed-off-by: Thierry Reding <treding@nvidia.com>
Reviewed-by: Alex Deucher <alexander.deucher@amd.com>
Signed-off-by: Dave Airlie <airlied@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/drm_dp_mst_topology.c b/drivers/gpu/drm/drm_dp_mst_topology.c
index 6bc16d7..076a51c 100644
--- a/drivers/gpu/drm/drm_dp_mst_topology.c
+++ b/drivers/gpu/drm/drm_dp_mst_topology.c
@@ -1486,10 +1486,10 @@ static int drm_dp_send_enum_path_resources(struct drm_dp_mst_topology_mgr *mgr,
  return 0;
 }
 
-int drm_dp_payload_send_msg(struct drm_dp_mst_topology_mgr *mgr,
-       struct drm_dp_mst_port *port,
-       int id,
-       int pbn)
+static int drm_dp_payload_send_msg(struct drm_dp_mst_topology_mgr *mgr,
+       struct drm_dp_mst_port *port,
+       int id,
+       int pbn)
 {
  struct drm_dp_sideband_msg_tx *txmsg;
  struct drm_dp_mst_branch *mstb;
@@ -1540,10 +1540,10 @@ static int drm_dp_create_payload_step1(struct drm_dp_mst_topology_mgr *mgr,
  return 0;
 }
 
-int drm_dp_create_payload_step2(struct drm_dp_mst_topology_mgr *mgr,
-    struct drm_dp_mst_port *port,
-    int id,
-    struct drm_dp_payload *payload)
+static int drm_dp_create_payload_step2(struct drm_dp_mst_topology_mgr *mgr,
+           struct drm_dp_mst_port *port,
+           int id,
+           struct drm_dp_payload *payload)
 {
  int ret;
  ret = drm_dp_payload_send_msg(mgr, port, id, port->vcpi.pbn);
@@ -1553,10 +1553,10 @@ int drm_dp_create_payload_step2(struct drm_dp_mst_topology_mgr *mgr,
  return ret;
 }
 
-int drm_dp_destroy_payload_step1(struct drm_dp_mst_topology_mgr *mgr,
-     struct drm_dp_mst_port *port,
-     int id,
-     struct drm_dp_payload *payload)
+static int drm_dp_destroy_payload_step1(struct drm_dp_mst_topology_mgr *mgr,
+     struct drm_dp_mst_port *port,
+     int id,
+     struct drm_dp_payload *payload)
 {
  DRM_DEBUG_KMS("\n");
  /* its okay for these to fail */
@@ -1569,9 +1569,9 @@ int drm_dp_destroy_payload_step1(struct drm_dp_mst_topology_mgr *mgr,
  return 0;
 }
 
-int drm_dp_destroy_payload_step2(struct drm_dp_mst_topology_mgr *mgr,
-     int id,
-     struct drm_dp_payload *payload)
+static int drm_dp_destroy_payload_step2(struct drm_dp_mst_topology_mgr *mgr,
+     int id,
+     struct drm_dp_payload *payload)
 {
  payload->payload_state = 0;
  return 0;
-- 
1.7.1